### PR TITLE
Fixes #172 Get hostname from env vars

### LINKED ISSUE #172: hostname is undefined in mail sent
### This is a
- [x] Bug report
- [ ] Feature request

### Current behaviour

An email is sent when the user generates documentation. In that email there are links to `Preview`, `Download` and `Deploy` documentation. Unfortunately  when a link is clicked, we are lead to `http://undefined/preview/........` etc.

### PR DESCRIPTION
**Pull request by imujjwal96**
## Description

Tried getting hostname through various means, but couldn't find a solution unless took reference of Open-even where they've used something similar.
## Related Issue

Issue #172 
## Types of changes

- [x] Bug fix 
- [ ] New feature

## Checklist:

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only one commit per issue.

### CODE REVIEW

File: routes/index.js [hunk 1]
```diff
@@ -5,7 +5,6 @@ var crypter = require("../util/crypter.js")
 var validation = require("../public/scripts/validation.js");
 /* GET home page. */
 router.get("/", function(req, res, next) {
-  exports.hostname = req.get('host');
```
Comment (pri22296) [inline routes/index.js]: did you try req.headers.get('host')?

Comment (imujjwal96) [inline routes/index.js]: Both of them works fine. The problem is in exporting it to that file. Tried but couldn't find the solution for that

Comment (pri22296) [inline routes/index.js]: At the end of this file module.exports is being set to router. Does that override `exports.hostname`? not sure about this.

Comment (sch00lb0y) [inline routes/index.js]: instead of exporting. try 
this
`process.env.hostname = req.get('host');`

File: backend/mailer.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ var exports = module.exports = {};
 
 var nodemailer = require('nodemailer');
 var sgTransport = require('nodemailer-sendgrid-transport');
-var hostname =require('../routes/index').hostname;
+var hostname =process.env.HOSTNAME || 'yaydoc.herokuapp.com';
```
Comment (yukiisbored) [inline backend/mailer.js]: Is this meant for just Heroku? because `HOSTNAME` isn't a POSIX thing.

Comment (imujjwal96) [inline backend/mailer.js]: It is yet another environment variable which can be set either in the .env file or as a config var in Heroku.

File: routes/index.js (hunk 1 above)
Comment (imujjwal96) [inline routes/index.js]: didn't work

**Comment by pri22296** [comment]
@imujjwal96 status of this PR? Is this good to go?

**Comment by imujjwal96** [comment]
It has been good to go since that day! Still waiting for your reviews.

**Comment by pri22296** [comment]
Sorry for that.

File: backend/mailer.js (hunk 2 above)
Comment (pri22296) [inline backend/mailer.js]: should HOSTNAME be also defined in app.json?

Comment (imujjwal96) [inline backend/mailer.js]: Updated.

**Comment by imujjwal96** [comment]
Can somebody please explain what is the issue with this PR? Why is it not been approved yet?

**Comment by pri22296** [comment]
@sch00lb0y Could you please review this. @imujjwal96 Please resolve merge conflicts

**Pull request merged**